uaclient/entitlements/fips.py

```diff
@@ -236,13 +290,21 @@ def _replace_metapackage_on_cloud_instance(
         ]
 
     @property
-    def packages(self) -> List[str]:
+    def packages(self) -> Optional[List[str]]:
+        if util.is_container():
+            return []
         packages = super().packages
-        return self._replace_metapackage_on_cloud_instance(packages)
+        if packages is not None:
+            return self._replace_metapackage_on_cloud_instance(packages)
+        return None
 
     def application_status(self) -> Tuple[status.ApplicationStatus, str]:
         super_status, super_msg = super().application_status()
 
+        if util.is_container() and not util.should_reboot():
+            self.cfg.remove_notice("", status.MESSAGE_FIPS_REBOOT_REQUIRED)
```

Comment:
Do we create this notice when enabling FIPS on a container ? Maybe we should instead restrict creating that notice during enable instead of cleaning it up here

Reply:
The problem is that `should_reboot` returns true after enabling fips in a container. I didn't look into it closely, but it made sense to me at the time that upgrading libssl might require services to restart (even if it technically doesn't require a system reboot).